### PR TITLE
Test: Fix unit test not building

### PR DESCRIPTION
**Pull request by alfred2g**
Fix Unit Test not building

Description
-----------
Fix the build framework for the UT, transport, and ble

Checklist:
----------

- [ X ] I have tested my changes. No regression in existing tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/ble/include/iot_ble_mqtt_serialize.h [hunk 1]
```diff
@@ -32,6 +33,8 @@
 #ifndef IOT_BLE_MQTT_SERIALIZE_H
 #define IOT_BLE_MQTT_SERIALIZE_H
 
+#define bool    int
```
Comment (aggarw13) [inline libraries/c_sdk/standard/ble/include/iot_ble_mqtt_serialize.h:36]: This seems like a dangerous change... It wouldn't work for compilers that have C99 support.

Comment (alfred2g) [reply libraries/c_sdk/standard/ble/include/iot_ble_mqtt_serialize.h:36]: bool won't work with c89 compilers 
so we need to fix all this file

File: libraries/c_sdk/standard/ble/utest/ble_mqtt_transport/CMakeLists.txt [hunk 2]
```diff
@@ -64,6 +66,8 @@
             "${AFR_ROOT_DIR}/freertos_kernel/include/"
             "${AFR_ROOT_DIR}/libraries/c_sdk/standard/common/include"
             "${AFR_ROOT_DIR}/libraries/coreMQTT/source/include"
+            "${AFR_ROOT_DIR}/libraries/coreMQTT/test/unit-test"
```
Comment (aggarw13) [inline libraries/c_sdk/standard/ble/utest/ble_mqtt_transport/CMakeLists.txt:69]: Why does this need to have the `coreMQTT/test/unit-test` path in the include list?

File: libraries/abstractions/transport/CMakeLists.txt [hunk 3]
```diff
@@ -35,4 +35,3 @@ afr_module_include_dirs(
        "${transport_interface_dir}"
        "${src_dir}"
 )
-
```
Comment (aggarw13) [inline libraries/abstractions/transport/CMakeLists.txt:38]: We usually do put newlines at the end of files...

**Review by aggarw13 — DISMISSED** [review]

**Review by aggarg — DISMISSED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**